### PR TITLE
DAOS-13141 control: Port daos cont create handlers to Go

### PR DESCRIPTION
**Pull request by kjacque**
The handlers for the daos cont create commands were still in C code, which can lead to races to print output.

Features: daos_cmd container

Required-githooks: true

### Before requesting gatekeeper:

* [x] Two review approvals and any prior change requests have been resolved.
* [x] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [x] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [x] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [x] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate watchers.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

File: src/control/cmd/daos/container.go [hunk 1]
```diff
@@ -281,102 +281,242 @@ func (cmd *containerCreateCmd) Execute(_ []string) (err error) {
...
 		if cmd.Mode.Set {
-			ap.mode = cmd.Mode.Mode
+			attr.da_mode = cmd.Mode.Mode
 		}
 		if cmd.CHints != "" {
-			ap.hints = C.CString(cmd.CHints)
-			defer freeString(ap.hints)
+			hint := C.CString(cmd.CHints)
+			defer freeString(hint)
+			C.strncpy(&attr.da_hints[0], hint, C.DAOS_CONT_HINT_MAX_LEN-1)
```
Comment (knard-intel) [inline src/control/cmd/daos/container.go:360]: Output of `strncpy()` call should be checked. From what I understand, If it is equal to ` C.DAOS_CONT_HINT_MAX_LEN-1`, the input string will be truncated and could be not null-terminated.
Same remark for the function `contCreateUNS()`.

**Review by knard-intel — COMMENTED** [review]
Mostly OK for me.

**Review by tanabarr — APPROVED** [review]

File: src/control/cmd/daos/container.go (hunk 1 above)
Comment (kjacque) [reply src/control/cmd/daos/container.go:360]: Go should be automatically zeroing structures, so I would expect the da_hints field to be all zeroes. If the input gets truncated, the last byte will be a zero.

**Comment by kjacque** [comment]
The rebuild/aggregate EC failure is [DAOS-12883](https://daosio.atlassian.net/browse/DAOS-12883).

File: src/control/cmd/daos/container.go [hunk 2]
```diff
@@ -281,102 +281,242 @@ func (cmd *containerCreateCmd) Execute(_ []string) (err error) {
...
+
+	var ci *containerInfo
+	ci, err = cmd.queryContainer()
+	if err != nil {
+		if errors.Cause(err) != daos.NoPermission {
+			return errors.Wrapf(err, "failed to query new container %s", contID)
+		}
+
+		// Special case for creating a container without permission to query it.
+		cmd.Errorf("container %s was created, but query failed", contID)
```
[the remark this answers is not in the document]
Comment (kjacque) [reply src/control/cmd/daos/container.go:307]: I'll change this if I have to make any other changes.

File: src/control/cmd/daos/container.go (hunk 1 above)
Comment (knard-intel) [reply src/control/cmd/daos/container.go:360]: Sounds logical for the zero padding.
NIT, do you think some Notice message should be done to indicate that the hint string has been truncated ?

**Review by knard-intel — APPROVED** [review]

**Pull request merged**